src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/UnsupportedStatement.scala

```diff
@@ -0,0 +1,49 @@
+package net.snowflake.spark.snowflake.pushdowns.querygeneration
+
+import net.snowflake.spark.snowflake._
+import org.apache.spark.sql.catalyst.expressions.{
+  Attribute,
+  Expression,
+  PythonUDF,
+  ScalaUDF
+}
+import org.apache.spark.sql.execution.aggregate.ScalaUDAF
+
+/**
+  * This class is used to catch unsupported statement and raise an exception
+  * to stop the push-down to snowflake.
+  */
+private[querygeneration] object UnsupportedStatement {
+
+  final val EXCEPTION_MESSAGE = "Meet unsupported statement"
+
+  /** Used mainly by QueryGeneration.convertStatement. This matches
+    * a tuple of (Expression, Seq[Attribute]) representing the expression to
+    * be matched and the fields that define the valid fields in the current expression
+    * scope, respectively.
+    *
+    * @param expAttr A pair-tuple representing the expression to be matched and the
+    *                attribute fields.
+    * @return An option containing the translated SQL, if there is a match, or None if there
+    *         is no match.
+    */
+  def unapply(
+               expAttr: (Expression, Seq[Attribute])
+             ): Option[SnowflakeSQLStatement] = {
+    val expr = expAttr._1
+
+    throw new SnowflakePushdownUnsupportedException(
+      EXCEPTION_MESSAGE,
+      expr.prettyName,
+      expr.sql,
+      isKnownUnsupportedOperation(expr))
+  }
+
+  // Determine whether the unsupported operation is known or not.
+  private def isKnownUnsupportedOperation(expr: Expression): Boolean = {
+    // The pushdown for UDF is known unsupported
```

Comment:
I'm actually not sure if we want to filter these out. Maybe we can flag them, but these are useful also to capture for statistics. For example, we might want to know how many customers are trying to pushdown UDFs.

Reply:
This is optional. I decided to filter them out because our purpose is to catch unsupported statement and support them in the future. The message for known unsupported statement sounds noise for this purpose.
If we need to know how many query are using UDF, I think we can query "spark_plan" to get it.